official/utils/logging/hooks.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Hook that counts examples per second every N steps or seconds."""
```

Comment:
Not sure if "examples" is the correct terminology, should it be "data" or a more accurate term?

Reply:
In this case, examples is actually more clear-- in training, we call each record/image/whatever an "example," and while a steps per second counter exists in TF, this one is counting over all GPUs how many actual examples have been processed (ie, steps * batch_size).